src/main/java/com/couchbase/lite/support/action/Action.java

```diff
@@ -0,0 +1,342 @@
+/**
+ * Created by Pasin Suriyentrakorn on 8/29/15.
+ * <p/>
+ * Copyright (c) 2015 Couchbase, Inc All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.couchbase.lite.support.action;
+
+import com.couchbase.lite.Misc;
+import com.couchbase.lite.support.FileDirUtils;
+import com.couchbase.lite.util.Log;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * An AtomicAction implementation to provide an ability to perform actions,
+ * back them out if there is an error, and clean up.
+ */
+public class Action implements AtomicAction {
```

Comment:
Thanks! I have ported the test on my personal mac but forgot to move it. Will do it.